### PR TITLE
[122] add lang strings and fix notices

### PR DESCRIPTION
**Pull request by hdagheda**
closes !122

### CODE REVIEW

**Comment by hdagheda** [comment]
```
root@b93f79f7407f:/var/www/mdl-39-stable# vendor/bin/phpunit admin/tool/heartbeat/tests/lib_test.php               
Moodle 3.9.16 (Build: 20220822), fd0252440b04c83da9d3f4670ca04f17544372f8
Php: 7.4.28, pgsql: 9.6.19, OS: Linux 5.15.0-47-generic x86_64
PHPUnit 7.5.20 by Sebastian Bergmann and contributors.

.                                                                   1 / 1 (100%)

Time: 1.09 seconds, Memory: 62.50 MB

OK (1 test, 4 assertions)
root@b93f79f7407f:/var/www/mdl-39-stable# vendor/bin/phpunit admin/tool/heartbeat/tests/check/tasklatencycheck_test.php 
Moodle 3.9.16 (Build: 20220822), fd0252440b04c83da9d3f4670ca04f17544372f8
Php: 7.4.28, pgsql: 9.6.19, OS: Linux 5.15.0-47-generic x86_64
PHPUnit 7.5.20 by Sebastian Bergmann and contributors.

...                                                                 3 / 3 (100%)

Time: 877 ms, Memory: 62.50 MB

OK (3 tests, 11 assertions)
```

File: classes/check/tasklatencycheck.php [hunk 1]
```diff
@@ -53,89 +53,103 @@ public function get_result(): result {
...
 
         $tasks = explode(PHP_EOL, $taskconfig);
-        foreach ($tasks as $taskconfig) {
-            list($taskclass, $runtime, $startdelay, $completiondelay) = explode(',', str_replace(' ', '', $taskconfig));
-            $task = \core\task\manager::get_scheduled_task($taskclass);
 
-            // Input validation.
-            $valid = $task !== false;
+        foreach ($tasks as $taskconfig) {
+            if ($taskconfig) {
```
Comment (Peterburnett) [inline classes/check/tasklatencycheck.php:58]: This seems weird, what is it trying to catch here? If its just an empty line, I think instead we should just trim/validate more heavily beforehand.

File: classes/check/tasklatencycheck.php [hunk 2]
```diff
@@ -53,89 +53,103 @@ public function get_result(): result {
...
+                list($taskclass, $runtime, $startdelay, $completiondelay) = explode(',', str_replace(' ', '', $taskconfig));
```
Comment (Peterburnett) [inline classes/check/tasklatencycheck.php:59]: This is the main line throwing errors, so perhaps I shouldnt have assumed the count. So instead, here we could do the explode, count the elements = 4, throw the error if not, and if there is 4, do the  list assignment

```
$configarr = explode(',', str_replace(' ', '', $taskconfig));
if (count($configarr) !== 4) {
  return new result(result::ERROR, get_string('taskconfigbad', 'tool_heartbeat', $taskclass));
}
list($taskclass, $runtime, $startdelay, $completiondelay) = $configarr;
```

File: classes/check/tasklatencycheck.php [hunk 3]
```diff
@@ -53,89 +53,103 @@ public function get_result(): result {
...
+
+                    if ($record && $record->duration > $runtime * MINSECS) {
+                        $status = result::CRITICAL;
+                        $messages .= get_string(
+                            'latencyruntime',
+                            'tool_heartbeat',
+                            ['task' => $taskclass, 'mins' => $startdelay]
+                        ) . PHP_EOL;
+                    }
+                } elseif ($lockstats) {
```
Comment (Peterburnett) [inline classes/check/tasklatencycheck.php:136]: `elseif` here is causing the CI failure

File: classes/check/tasklatencycheck.php [hunk 4]
```diff
@@ -111,31 +124,36 @@ public function get_result(): result {
             if ($dbman->table_exists($table)) {
                 // We can use task logs!
                 $sql = "SELECT (timeend - timestart) AS duration
-                            FROM {task_log}
-                            WHERE classname = ?
-                        ORDER BY timeend DESC
-                            LIMIT 1";
+                                    FROM {task_log}
```
Comment (Peterburnett) [inline classes/check/tasklatencycheck.php:127]: Could you quickly tweak these lines to be back inline with the top line?

File: classes/check/tasklatencycheck.php [hunk 5]
```diff
@@ -111,31 +124,36 @@ public function get_result(): result {
...
                 }
-
             } else if ($lockstats) {
                 // Our fallback here is lockstats.
                 $sql = "SELECT duration
-                            FROM {tool_lockstats_history}
-                            WHERE classname = ?
-                        ORDER BY released DESC
-                            LIMIT 1";
+                                    FROM {tool_lockstats_history}
```
Comment (Peterburnett) [inline classes/check/tasklatencycheck.php:144]: Same as above

**Pull request merged**